pkg/controllers/updaterun/execution_integration_test.go

```diff
@@ -702,6 +702,551 @@ var _ = Describe("UpdateRun execution tests - double stages", func() {
+			wantStatus.StagesStatus[1].Conditions[0] = generateFalseCondition(updateRun, placementv1beta1.StageUpdatingConditionProgressing) // The progressing condition now becomes false with waiting reason.
+			wantStatus.StagesStatus[1].AfterStageTaskStatus[0].Conditions = append(wantStatus.StagesStatus[1].AfterStageTaskStatus[0].Conditions,
+				generateTrueCondition(updateRun, placementv1beta1.StageTaskConditionApprovalRequestCreated))
+			meta.SetStatusCondition(&wantStatus.Conditions, generateFalseCondition(updateRun, placementv1beta1.StagedUpdateRunConditionProgressing))
+			validateClusterStagedUpdateRunStatus(ctx, updateRun, wantStatus, "")
+
+			By("Checking update run status metrics are emitted")
+			// Remove any existing waiting of the same generation metric and add to the end.
+			waitingMetric := generateWaitingMetric(updateRun)
+			wantMetrics = removeMetricFromMetricList(wantMetrics, waitingMetric)
+			wantMetrics = append(wantMetrics, waitingMetric)
+			validateUpdateRunMetricsEmitted(wantMetrics...)
+		})
+
+		It("Should approve request for 2nd stage after", func() {
+			By("Validating the approvalRequest has been created")
+			wantApprovalRequest := &placementv1beta1.ClusterApprovalRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: updateRun.Status.StagesStatus[1].AfterStageTaskStatus[0].ApprovalRequestName,
+					Labels: map[string]string{
+						placementv1beta1.TargetUpdatingStageNameLabel:   updateRun.Status.StagesStatus[1].StageName,
+						placementv1beta1.TargetUpdateRunLabel:           updateRun.Name,
+						placementv1beta1.IsLatestUpdateRunApprovalLabel: "true",
+					},
+				},
+				Spec: placementv1beta1.ApprovalRequestSpec{
+					TargetUpdateRun: updateRun.Name,
+					TargetStage:     updateRun.Status.StagesStatus[1].StageName,
+				},
+			}
+			validateApprovalRequestCreated(wantApprovalRequest)
+
+			By("Approving the approvalRequest")
+			approveClusterApprovalRequest(ctx, wantApprovalRequest.Name)
+		})
+
+		It("Should start pausing the update run when state is Stop", func() {
+			By("Updating updateRun state to Stop")
+			updateRun.Spec.State = placementv1beta1.StateStop
+			Expect(k8sClient.Update(ctx, updateRun)).Should(Succeed(), "failed to update the updateRun state")
+
+			By("Validating the update run is stopped")
+			meta.SetStatusCondition(&wantStatus.Conditions, generateFalseConditionWithReason(updateRun, placementv1beta1.StagedUpdateRunConditionProgressing, condition.UpdateRunStoppedReason))
+			validateClusterStagedUpdateRunStatus(ctx, updateRun, wantStatus, "")
+
+			By("Checking update run status metrics are emitted")
+			wantMetrics = append(wantMetrics, generateStoppedMetric(updateRun))
+			validateUpdateRunMetricsEmitted(wantMetrics...)
+		})
+
+		It("Should not continue rolling 2nd stage when stopped", func() {
+			By("Validating update run is stopped")
+			validateClusterStagedUpdateRunStatus(ctx, updateRun, wantStatus, "")
```

Comment:
This seems like an ambiguous status check, why do we approve the approvalRequest and then make this check I'm not sure if my status will remain the same before we validate the status. 

What if the test is slow and the controller reconciles the updateRun and updates status before we check status ?

Instead we can stop before approval, approve the request and then check if updateRun status is not updated

We can do the same before deleteStage